### PR TITLE
Add release instructions

### PR DESCRIPTION
**Pull request by jhale**
Probably won't get all of this done in one go but it's a start.

### CODE REVIEW

**Comment by jhale** [comment]
The release branches are in a reasonable state for finalising a release.

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+install the `regclient` utility https://github.com/regclient/regclient using the
+provided binaries. `regclient` can properly handle copying multi-architecture images.
+
+    regctl registry login docker.io
+    regctl image copy dolfinx/dolfinx:<tag> dolfinx:dolfinx:stable 
+    regctl image copy dolfinx/lab:<tag> dolfinx/lab:stable 
+    regctl image copy dolfinx/dev-env:<tag> dolfinx/dev-env:stable 
+    regctl image copy dolfinx/dolfinx-onbuild:<tag> dolfinx/dolfinx-onbuild:stable
+
+### pypa
```
Comment (jorgensd) [inline RELEASE.md:155]:
```suggestion
### pypi
```

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+### FFCx version bump
+
+1. Merge `main` into `release` resolving all conflicts in favour of `main`.
+
+       git merge -Xtheirs main
+
+2. Update the version number in `setup.cfg`, e.g. `0.5.0`.
+
+3. Update the dependency versions for `fenics-basix` and `fenics-ufl` in `setup.cfg`.
+
```
Comment (jorgensd) [inline RELEASE.md:70]: @chrisrichardson  was there a step missing here?

**Comment by chrisrichardson** [comment]
Another thing to mention, is that whilst the integration testing worked smoothly, the tags did not build the documentation the first time, because the CI was pointing to `main` and the compatibility requirement failed. I changed the CI to look at `release` branch for the dependencies, then it worked.
It may be better (perhaps) to update the versions in `main` before tagging?

**Comment by jhale** [comment]
Thanks for the explanation! I see the issue, I'll add something to the documentation. 

In the content of DOLFINx, it would be great if the Actions simply read a JSON file with the git refs of the accompanying FEniCSx components, rather than having to change them manually in about 10 files.

**Comment by nate-sime** [comment]
Should some details about adding a new version to the Spack packages be added in a future PR?

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+6. Check `git diff main` for obvious errors.
+
+## Integration testing
+
+Although lengthy, integration testing is highly effective at discovering issues
+and mistakes before they reach tagged versions.
+
+At each of the following links run the GitHub Action Workflow manually using
+the `develop` branch in all fields. *Only proceed to tagging once all tests pass.*
```
Comment (garth-wells) [inline RELEASE.md:108]: We have `develop` branches?

**Pull request merged**